_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,325 @@
+open PR’s. 	
+
+![bundlephobia Enzyme sreen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/04.png)
+###### Source [Bundle Phobia](https://bundlephobia.com/package/enzyme@3.11.0)
+
+
+Interesting fact: 
+- There are also Enzyme Adapter Packages for React. So you need to install adapter package and its dependencies related 
+to the version of React that you are using. For example, for version 16, it also needs *enzyme—adapter—react—16* to be 
+installed. Unfortunately, it does not support all features completely. So for every new version of React, we have to 
+wait for the release of the new adapter.
+
+## React Testing Library (often called RTL)
+
+RTL says about itself: *The React Testing Library is a very light—weight solution for testing React components. It 
+provides light utility functions on top of react—dom and react—dom/test—utils, in a way that encourages better testing 
+practices.*.
+
+Released in **April 2018 by Kent C. Dodds**. The latest stable version **12.1.2** of RTL has a bundle size of 
+**215.8kb (47.3 kB)** and was released in **October 2021**. Currently, a pre—release of v13.0.0—alpha is being 
+prepared. Repository has 26 [open issues](https://github.com/enzymejs/enzyme/issues) and 2 open PR’s.  
+
+![bundlephobia RTL screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/05.png)
+
+###### Source [Bundle Phobia](https://bundlephobia.com/package/@testing-library/react@12.1.2)
+
+
+Interesting facts: 
+- When you create a project by Create React App, you will get support for React Testing Library right away. 
+- Jest documentation links to the RTL website in the [*Testing Web Framework*](https://jestjs.io/docs/testing-frameworks#react)
+ section as well as React documentation in the [*Recommended Tools*](https://reactjs.org/docs/testing.html#tools).
+- Testing—library allows us to test with the same API not only React but also others frameworks like Vue 
+(testing—library/vue), Angluar (tesiting—library/angular), Svelte (testing—library/svelte) etc. This could be some 
+advantage for people working on several projects in different technologies.
+
+## Basic differences
+
+The main differences between the two libraries that I can notice:
+1. **Different approaches to testing** 
+   
+   What distinguishes RTL is that we find elements by their labels or texts, almost like a user would. The point is to 
+   replicate and simulate the user’s behavior. Rtl tries to render components to the DOM and doesn’t give us access 
+   directly to the internal component. While Enzyme uses shallow rendering (or sometimes deep rendering) and 
+   encourages us to test the components instance using their state and props. Enzyme also allows full rendering of DOM 
+   but it is not the basic assumption of the library’s operation.
+2. **Bundle size** 
+   
+   Maybe this is not as significant as the first point, but it is worth paying attention to bundle size also. RTL has 
```

Comment:
I don't think the size matters that much, it doesn't impact the bundle size of an app.